src/main/java/Parser.java

```diff
@@ -0,0 +1,108 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Parser takes in a variety of strings, makes sense of the strings and creates relevant objects.
+ */
+public class Parser {
+	/**
+	 * Parses dates that are of the accepted date formats.
+	 *
+	 * @param str String to parse.
+	 * @return Date if successfully parsed, null if wrong date format.
+	 */
+	public static Date parseDate(String str) {
+		List<String> formatStrings = Arrays.asList("yyyy-M-dd", "dd/M/yyyy HHmm", "dd/M/yyyy", "MMM d yyyy");
+
+		for (String formatString : formatStrings) {
+			try {
+				return new SimpleDateFormat(formatString).parse(str);
+			} catch (ParseException e) {}
+		}
```

Comment:
Is the catch statement supposed to be empty here? If so, perhaps a comment explaining why this is empty would help.